pkg/shell/base_index.js

```diff
@@ -940,8 +941,20 @@ function CompiledComponents() {
                     label: cockpit.gettext(info.label) || prop,
                     order: info.order === undefined ? 1000 : info.order,
                     icon: info.icon,
-                    wants: info.wants
+                    wants: info.wants,
+                    keywords: info.keywords || [{ matches: [] }],
+                    keyword: { score: -1 }
                 };
+
+                // Always first keyword should be page name
+                item.keywords[0].matches.unshift(item.label.toLowerCase());
+
+                // Keywords from manifest have different defaults than are usual
+                item.keywords.forEach(i => {
+                    i.weight = i.weight || 3;
+                    i.translate = i.translate || true;
```

Comment:
This doesn't work as intended, as `false || true` is still true. You need a `=== undefined` here, I suppose.